File: practice13/14.cpp
//https://www.nowcoder.com/practice/637062df51674de8ba464e792d1a0ac6?tpId=37&tqId=21319&tPage=5&rp=&ru=/ta/huawei&qru=/ta/huawei/question-ranking
#include<iostream>
#include<string>
using namespace std;
int main()
{
	string str;
	string ch = "*";
	char p;
	int m, i, sign;
	while (cin >> str)
	{
		m = str.size();
		sign = 0;
		for (i = 0; i < str.size() + 1; i++)
		{
			p = str[i];
			if (str[i] >= '0'&&str[i] <= '9'&&sign == 0)
			{
				str.insert(i, ch);
				sign = 1;
			}
			else if (sign == 1 && (str[i]<'0' || str[i]>'9'))
			{
				sign = 0;
				str.insert(i, ch);
			}
		}
		cout << str << endl;
	}
	system("pause");
	return 0;
}

File: practice31/14.java




//https://www.nowcoder.com/practice/6e6ad6338289498da79b7afb60e823b3?tpId=90&tqId=30894&tPage=6&rp=6&ru=/ta/2018test&qru=/ta/2018test/question-ranking
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String A = sc.nextLine();
        String B = sc.nextLine();
        StringBuilder sb = new StringBuilder();
        for(int i=0;i<B.length();i++){
            if(B.charAt(i)=='?'){
                sb.append("[01]");
            }
            else{
                sb.append(B.charAt(i));
            }
        }
        Set<String> set=new HashSet<String>();
        for(int i=0;i<=A.length()-B.length();i++){
            String sub=A.substring(i,i+B.length());
            if(sub.matches(sb.toString())){
                set.add(sub);
            }
        }
        System.out.println(set.size());
    }
}